finagle-netty4/src/main/scala/com/twitter/finagle/netty4/channel/Netty4ServerChannelInitializer.scala

```diff
@@ -103,13 +104,25 @@ private[netty4] class Netty4ServerChannelInitializer(
 
     pipeline.addLast("exception handler", exceptionHandler)
 
+    // Need to retain the ByteBuf before it's forwarded to another thread
+    pipeline.addLast("ContentRetainer", ContentRetainer)
+
     // The bridge handler must be last in the pipeline to ensure
     // that the bridging code sees all encoding and transformations
     // of inbound messages.
     pipeline.addLast("finagle bridge", newBridge())
   }
 }
 
+private[netty4] object ContentRetainer extends ChannelInboundHandlerAdapter {
```

Comment:
Ah that's interesting! I guess you're right and we should totally ratain those objects.

What do you think about fixing `ChannelTransport` to do `ReferenceCountUtil.ratain(msg)` before offering the message into `AsyncQueue`, instead of introducing a new handler here?